deployments/helm/gpud/templates/daemonset.yaml

```diff
@@ -96,31 +150,132 @@ spec:
+              set -o nounset
+              set -o pipefail 2>/dev/null || true  # pipefail not supported in dash (Ubuntu's /bin/sh)
+
+              # Node label file (if nodeLabelExporter sidecar is enabled)
+              LABEL_FILE="{{ .Values.nodeLabelExporter.outputFile }}"
+
+              # Label keys (can be overridden via env)
+              EP_LABEL_KEY="${GPUD_ENDPOINT_LABEL_KEY:-{{ .Values.nodeLabelExporter.labelKeys.endpoint }}}"
+              TOKEN_LABEL_KEY="${GPUD_TOKEN_LABEL_KEY:-{{ .Values.nodeLabelExporter.labelKeys.token }}}"
+              MID_LABEL_KEY="${GPUD_MACHINE_ID_LABEL_KEY:-{{ .Values.nodeLabelExporter.labelKeys.machineId }}}"
+              EP=""
+              TOKEN=""
+              MID=""
+              if [ -f "$LABEL_FILE" ]; then
+                # Use awk with index() for exact string matching to:
+                # 1. Avoid regex issues (dots/slashes in keys are literal)
+                # 2. Preserve '=' characters in values (important for base64 tokens)
+                EP="$(awk -v key="$EP_LABEL_KEY" 'index($0, key"=") == 1 {print substr($0, length(key)+2); exit}' "$LABEL_FILE")"
+                TOKEN="$(awk -v key="$TOKEN_LABEL_KEY" 'index($0, key"=") == 1 {print substr($0, length(key)+2); exit}' "$LABEL_FILE")"
+                MID="$(awk -v key="$MID_LABEL_KEY" 'index($0, key"=") == 1 {print substr($0, length(key)+2); exit}' "$LABEL_FILE")"
+              fi
+
+              # Build final command arguments using set -- for safety (avoids eval and quoting issues)
+              set -- run
+              set -- "$@" --listen-address="{{ .Values.gpud.listenAddress }}"
+              set -- "$@" --log-level="{{ .Values.gpud.logLevel }}"
+
+              # Determine endpoint (label takes precedence over values.yaml)
+              ENDPOINT="{{ .Values.gpud.endpoint }}"
+              if [ -n "$EP" ]; then
+                ENDPOINT="$EP"
+              fi
+              if [ -n "$ENDPOINT" ]; then
+                set -- "$@" --endpoint="$ENDPOINT"
+              fi
+              if [ -n "$TOKEN" ]; then
+                set -- "$@" --token="$TOKEN"
+              fi
+              if [ -n "$MID" ]; then
+                set -- "$@" --machine-id="$MID"
+              fi
+
+              set -- "$@" --enable-auto-update={{ .Values.gpud.enableAutoUpdate }}
+              set -- "$@" --auto-update-exit-code={{ .Values.gpud.autoUpdateExitCode }}
+
+              echo "Starting: /usr/local/bin/gpud $*"
+              exec /usr/local/bin/gpud "$@"
 
           env:
             - name: GPUD_NO_USAGE_STATS
               value: {{ not .Values.gpud.telemetry.enabled | quote }}
 
           {{- if .Values.service.enabled }}
           ports:
-            - name: http
-              containerPort: {{ .Values.service.port }}
+            - name: https
+              # Must match the port in gpud.listenAddress (default 15132)
+              containerPort: {{ .Values.livenessProbe.httpGet.port }}
```

Comment:
Nice! I'm glad you resolved this as well, as we needed to change `.Values.service.port` to 15132 to get a ServiceMonitor to work, before switching to PodMonitor.

What do you think about removing the `.Values.service.port` entirely if you're going to use the liveness probe port instead?

Reply:
On second thought, it might actually be more semantic to use `.Values.service.port` here but default it to port 15132.

The reason being that someone could define a livenessProbe in their `values.yaml` file that doesn't use an `httpGet` at all, and this would fail.